language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -565,6 +567,52 @@ writePackageIdTs dir pkgId =
     T.writeFileUtf8 (dir </> "packageId.ts") $ T.unlines
       ["export default '" <> unPackageId pkgId <> "';"]
 
+writeIndexTs :: FilePath -> [Module] -> IO ()
+writeIndexTs packageSrcDir modules =
+  T.writeFileUtf8 (packageSrcDir </> "index.ts") lines
+  where
+    lines :: T.Text
+    lines = T.unlines $ header <> concatMap entry modules
+
+    header :: [T.Text]
+    header =
+      [ "import {default as __packageId} from \"./packageId\""
+      , "export const packageId = __packageId"
+      ]
+
+    entry :: Module -> [T.Text]
+    entry mod
+      | null $ defDataTypes mod = []
+      | otherwise = importDecl var path <> exportDecl var parts
+      where
+        parts = unModuleName (moduleName mod)
+        var = "__" <> T.intercalate "_" parts
+        path =  "\"./" <> T.intercalate "/" parts <> "\""
+
+    importDecl :: T.Text -> T.Text -> [T.Text]
+    importDecl var path =
+      [ "/* eslint-disable @typescript-eslint/camelcase */"
+      , "import * as " <> var <> " from "  <> path
+      ]
+
+    exportDecl :: T.Text -> [T.Text] -> [T.Text]
+    exportDecl = go 0
+      where
+        spaces i = T.pack $ replicate i ' '
+
+        go :: Int -> T.Text -> [T.Text] -> [T.Text]
+        go indent var [m] = let ws = spaces indent in
+          [ ws <> "/* eslint-disable @typescript-eslint/no-unused-vars */"
+          , ws <> "export import " <> m <> " = " <> var
+          ]
+        go indent var (m : parts) = let ws = spaces indent in
+          [ ws <> "/* eslint-disable @typescript-eslint/no-namespace */"
+          , ws <> "export namespace " <> m <> " {"
+          ] <>
+          go (indent + 2) var parts <>
```

Comment:
I find the inlined pretty printing quite hard to read. Do we actually need this? Maybe leave it out for now and replace it by a proper pretty printing library at some point?